lib/common/schemaUtilsCommon.js

```diff
@@ -2,7 +2,207 @@
  * This file contains util functions that are common between versions
  */
 
-const parse = require('../parse.js');
+const parse = require('../parse.js'),
+  _ = require('lodash'),
+  typesMap = {
+    integer: {
+      int32: '<integer>',
+      int64: '<long>'
+    },
+    number: {
+      float: '<float>',
+      double: '<double>'
+    },
+    string: {
+      byte: '<byte>',
+      binary: '<binary>',
+      date: '<date>',
+      'date-time': '<dateTime>',
+      password: '<password>'
+    },
+    boolean: '<boolean>',
+    array: '<array>',
+    object: '<object>'
+  },
+
+  /* eslint-disable arrow-body-style */
+  schemaTypeToJsValidator = {
+    'string': (d) => typeof d === 'string',
+    'number': (d) => !isNaN(d),
+    'integer': (d) => !isNaN(d) && Number.isInteger(Number(d)),
+    'boolean': (d) => _.isBoolean(d) || d === 'true' || d === 'false',
+    'array': (d) => Array.isArray(d),
+    'object': (d) => typeof d === 'object' && !Array.isArray(d)
+  };
+
+/**
+ *  Remove the # character from the beginning of a schema path
+ * @param {string} schemaPath - a defined schemaPath
+ * @returns {string} - the schema path with # removed
+ */
+function removeSharpAndSlashFromFirstPosition(schemaPath) {
+  return schemaPath[0] === '#' ? schemaPath.slice(2) : schemaPath;
+}
+
+/**
+ *  Remove the defined word from the last position of a schema path
+ * @param {string} schemaPath - a defined schemaPath
+ * @param {string} word - word to remove
+ * @returns {string} - the schema path with type removed
+ */
+function removeWordFromLastPosition(schemaPath, word) {
+  let splittedDataPath = schemaPath.split('/');
+  if (splittedDataPath[splittedDataPath.length - 1] === word) {
+    splittedDataPath.splice(-1);
+  }
+  return splittedDataPath.join('/');
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ *
+ * @param {string} value - Value to check for
+ * @param {string} type - The type in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function compareType(value, type) {
+  return value === '<' + type + '>';
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ * Works in array types
+ * @param {string} value - Value to check for
+ * @param {*} types - The types in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function isTypeValueArrayCheck(value, types) {
+  return types.find((type) => {
+    return compareType(value, type);
+  }) !== undefined;
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ * Works in array types
+ * @param {string} value - Value to check for
+ * @param {*} types - The types in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function checkValueOnlyTypes(value, types) {
```

Comment:
Context: Above comment.
This function wouldn't be needed anymore if we reorganise this part of the code.